var/spack/repos/builtin/packages/libzip/package.py

```diff
@@ -51,9 +54,36 @@ def url_for_version(self, version):
         conditional("cmake", when="@1.4:"), conditional("autotools", when="@:1.3"), default="cmake"
     )
 
+    # Required dependencies
+    with when("build_system=cmake"):
+        variant("gnutls", default=True, description="Enable gnutls support")
+        variant("bzip2", default=True, description="Enable bzip2 support")
+        variant("lzma", default=True, description="Enable lzma support")
+        variant("openssl", default=True, description="Enable openssl support")
+        variant("zstd", default=True, description="Enable zstd support")
+        variant("mbedtls", default=True, description="Enable mbedtls support")
+        depends_on("gnutls", when="+gnutls")
+        depends_on("bzip2", when="+bzip2")
+        depends_on("lzma", when="+lzma")
+        depends_on("openssl", when="+openssl")
+        depends_on("zstd", when="+zstd")
+        depends_on("mbedtls", when="+mbedtls")
+
     @property
     def headers(self):
         # Up to version 1.3.0 zipconf.h was installed outside of self.prefix.include
         return find_all_headers(
             self.prefix if self.spec.satisfies("@:1.3.0") else self.prefix.include
         )
+
+
+class CMakeBuilder(spack.build_systems.cmake.CMakeBuilder):
+    def cmake_args(self):
+        return [
+            self.define_from_variant("ENABLE_GNUTLS", "gnutls"),
+            self.define_from_variant("ENABLE_MBEDTLS", "mbedtls"),
+            self.define_from_variant("ENABLE_OPENSSL", "openssl"),
+            self.define_from_variant("ENABLE_BZIP2", "bz2"),
```

Comment:
This variant does not exist. Its current name is "bzip2". You could not have built this recipe.
```py
==> Installing libzip-1.11.1-5aupui6t6j4ve3d6i3ffgslk3okyxkcf [28/29]
==> Error: KeyError: '"bz2" is not a variant of "libzip"'
```
What kind of tests do you run before you submitted this PR?

Reply:
yeah ok may be I was too fast to rename bz2 to bzip2 and didn't cheeck it now it is fixed in #47230

Reply:
I'd just let this PR wait for now.
(you can just close it, and re-open it again before you update it) until #47230 is merged.

It looks like can just remove the changes to `libzip` as this package does not depend on the new variants, which are the only change that is in that PR AFAICS.

So I wonder: Why you even add those variants?

Reply:
fmi4cpp doesn't care about the variants of libzip, I agree. However it is, IMHO, necessary to add them to libzip and properly define the relationship with them otherwise we get libzip depends from the host libs and from spack libs that libzip depend upon.